File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        bool flag=false;
        int n=nums.size();
        for(int i=0;i<n-1;i++){
            if(nums[i]<=nums[i+1]) continue;
            else if(nums[i]>nums[i+1] and flag==false){
                flag=true;
                if(nums[i]<nums[n-1] or nums[0]<nums[n-1]) return false;
            }
            else if(nums[i]>nums[i+1] and flag!=false) return false;
        }

        return true;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n=nums.length;
        int count=0;
        for(int i=0;i<n-1;i++){
            if(nums[i]<=nums[i+1]) continue;
            else count++;
            
        }
        if(count ==0) return true;
        else if(count==1 && nums[0]>=nums[n-1]) return true;
        else return false;
    }
}